packages/core/helper-plugin/lib/src/components/DynamicTable/TableHead/index.js

```diff
@@ -46,8 +46,8 @@ const TableHead = ({
           const isUp = sortOrder === 'ASC';
 
           const intlLabel = formatMessage({
-            id: `global.table.header.${name}`,
-            defaultMessage: label,
+            id: label || name || `global.table.header.${name}`,
```

Comment:
I was the one who merged @yangfei4913438 pull request and after rethinking about it I don't think the label translation should come from the helper-plugin component.
imo this fix will lead to lots of translations only based on `label` or `name` without any context (coming from the content-manager table, users table, etc).

in `content-manager/components/DynamicTable` I found an occurrence of having some context about the translation key for the content-manager table header:
`label: formatMessage({ id: getTrad('containers.ListPage.table-headers.published_at') })`
which makes it the only key that was translated so far.

I think this should be the way to go, then `helper-plugin/DynamicTable` wouldn't have to handle translation for the label.
to make it dynamic I would propose: 
`label: formatMessage({ id: getTrad(`containers.ListPage.table-headers.${name}`) })`

it would mean that when extending translation for the content manager table some context will be given to the translation, which I can understand is less simple than a simple `"Users"` key, but would prevent some conflict in having lots of translation keys without context in the long term.

this would mean some refactoring in different places which I started while finding a solution, I could create a PR for it unless you would want to handle it, wdyt?

Reply:
I agree with you! If you can create a PR,  I will cancel my PR.

Reply:
great :) we could leave this one open until I open a ready-to-review PR, I'll let you know when it is done!

Reply:
#14111 was merged today, it'll be out for the next release 🎉